src/tools/neorados.cc

```diff
@@ -367,10 +374,24 @@ int main(int argc, char* argv[])
 		   prog, command, ci->name, ci->usage);
 	return 1;
       }
-      s::spawn(c, [&](s::yield_context y) {
-		    auto r = R::RADOS::Builder{}.build(c, y);
-		    ci->f(r, parameters, y);
-		  });
+      ba::co_spawn(c,
+		   [&]() -> ba::awaitable<void> {
+		     try {
+		       auto r = co_await R::RADOS::Builder{}.build(
+			 c, ba::use_awaitable);
+		       co_await ci->f(r, parameters);
+		     } catch (const std::exception& e) {
+		       // Annoyingly, run does not throw coroutines
+		       // generated by `co_spawn`, so we have to catch
+		       // here and propagate the exit value, as well
+		       // as having the larger catch for any
+		       // exceptions thrown by `program_options` or
+		       // elsewhere.
+		       fmt::print(std::cerr, "{}: {}: {}\n", prog, command,
+				  e.what());
+		       exitval = EXIT_FAILURE;
+		     }
+		   }, ba::detached);
```

Comment:
according to https://www.boost.org/doc/libs/1_75_0/doc/html/boost_asio/reference/co_spawn/overload1.html, that last `ba::detached` argument could instead be a `handler(std::exception_ptr)`. i would expect that completion handler to run outside of the coroutine, so you could rethrow the exception there. `void std::rethrow_exception(std::exception_ptr p)` has that same signature, can you use it directly?
```c++
      ba::co_spawn(c,
		   [&]() -> ba::awaitable<void> {
		     auto r = co_await R::RADOS::Builder{}.build(
		         c, ba::use_awaitable);
		     co_await ci->f(r, parameters);
		   }, std::rethrow_exception);
```

Reply:
That works perfectly, thank you.